statServer/models/questions_model.ts

```diff
@@ -0,0 +1,88 @@
+/*
+El nuevo entrypoint tiene la siguiente estructura:
+
+GET /questions?since=2022-03-04
+
+{
+   "questionId": ".. ",  // Identificación de la pregunta 
+   "question": ".. ",  // Texto de la pregunta en español
+   "attempts":  #  // Número de veces que la pregunta se ha intentado 
+   "avgLatency": # // Tiempo promedio en segundo que la gente necesita para responder esta pregunta
+   "successProb": # //  La probabilidad de tener éxito respondiendo esta pregunta (valor entre cero y uno)
+}
+since: fecha para considerar los datos sólo después de la fecha dada, es opcional, si no está se consideran todos los datos disponibles.
+
+Para los textos de las preguntas se va a usar el resultado del issue Issue 54
+*/
+
+import { Questions } from '../types/questions';
+import { db } from "../db";
+import { RowDataPacket } from "mysql2";
+
+
+export const getQuetionsByDates = (callback: Function, dates: any) => {
+
+    let { since, to } = dates;
+    // console.log(since)
+    // console.log(to)
+
+    if (since === '' || to === '') {
+        console.log('cambio de since o to')
+        since = '01/01/0001'
+        to = '12/12/9999'
+    }
+
+    // console.log(since)
+    // console.log(to)
+    
+
+    console.log(dates)
+    const queryString = `
+    
+    SELECT
+        q.id AS questionId,
+        q.question,
+        COUNT(ua.question_id) AS attempts,
+        AVG(ua.latency / 1000) AS avgLatency,
+        SUM(ua.success) / COUNT(ua.question_id) AS successProb
+    FROM
+        question q
+    LEFT JOIN
+        user_answer ua ON q.id = ua.question_id
+    WHERE
+        CASE
+            WHEN EXISTS (
+                SELECT 1
+                FROM user_answer ua
+                WHERE DATE(FROM_UNIXTIME(ua.answered_on/1000)) BETWEEN STR_TO_DATE("${since}", '%d/%m/%Y') AND STR_TO_DATE("${to}", '%d/%m/%Y') 
+            )
+            THEN DATE(FROM_UNIXTIME(ua.answered_on/1000)) BETWEEN STR_TO_DATE("${since}", '%d/%m/%Y') AND STR_TO_DATE("${to}", '%d/%m/%Y')
```

Comment:
Este es un doble query a la DB, engorroso de indexar y muy largo para leer. Es mucho mejor:
    WHERE
        ua.answered_on BETWEEN UNIX_TIMESTAMP("${since}")*1000 AND UNIX_TIMESTAMP("${to}")*1000
de modo que podemos indexar la búsqueda de forma natural ( DATE(FROM_UNIXTIME(ua.answered_on/1000)) es indexable pero queda complicado) y si cambiamos el formato de la fecha de  %d/%m/%Y a  %Y-%d-%m la sintaxis es mucho más simple.

Reply:
Se realiza ajuste con base a lo sugerido.